bpf/init.sh

```diff
@@ -432,6 +434,9 @@ if [ "${TUNNEL_MODE}" != "<nil>" ]; then
 		ip link add name $ENCAP_DEV address $(rnd_mac_addr) type $TUNNEL_MODE external || encap_fail
 	}
 	ip link set $ENCAP_DEV mtu $MTU || encap_fail
+	if [ "${TUNNEL_PORT}" != "<nil>" ]; then
```

Comment:
I'm not sure if this is a problem in practice, but would it be better to set the `dstport` directly in the command above (`ip link add name $ENCAP_DEV address $(rnd_mac_addr) type $TUNNEL_MODE external`) rather than changing it after creating the device? We could have `$TUNNEL_PORT` always set (to either the default value or the user specified one) and save this if case/setting it to `<nil>`

Reply:
So actually the ideas is that port can change after device is created, so we need to cater for that. It maybe the case that `dstport` needs to be specified in both invocation, if we are worried about any race conditions here...

Reply:
> We could have `$TUNNEL_PORT` always set (to either the default value or the user specified one) and save this if case/setting it to `<nil>`

It is actually always going to be set, when `TUNNEL_MODE` is either `vxlan` or `geneve`, it's only set to `<nil>` when `TUNNEL_MODE` is neither of those two options... I this check is mostly for the sake of being explicit.

Reply:
> It is actually always going to be set

Sorry, what I meant with that was: always explicitly set to a specific port number (so that init.sh doesn't have to deal with the `<nil>` case)

Reply:
> so that init.sh doesn't have to deal with the `<nil>` case

This check is somewhat redundant, I could delete it now and behaviour shouldn't change. But as far as this script itself is concerned, `<nil>` case is still possible.